scrapper.py

```diff
@@ -1,63 +1,206 @@
-    # YOUR CODE HERE
-    pass
+"""
+Scrapper implementation
+"""
+from datetime import datetime, timedelta
+import json
+import re
+from pathlib import Path
+import random
+import shutil
+from time import sleep
+
+from bs4 import BeautifulSoup
+import requests
+
+from core_utils.article import Article
+from constants import ASSETS_PATH, CRAWLER_CONFIG_PATH, HTTP_PATTERN, HTTP_PATTERN1
+
+
+class IncorrectURLError(Exception):
+    """
+    Seed URL does not match standard pattern
+    """
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+    """
+    Total number of articles to parse is too big
+    """
+
+
+class IncorrectNumberOfArticlesError(Exception):
+    """
+    Total number of articles to parse in not integer
+    """
+
+
+class Crawler:
+    """
+    Crawler implementation
+    """
+
+    def __init__(self, seed_urls, total_max_articles):
+        self.seed_urls = seed_urls
+        self.total_max_articles = total_max_articles
+        self.urls = []
+
+    def _extract_url(self, article_bs):
+        articles_blocks = article_bs.find_all('div', class_='card-article__content')
+        not_full_urls = []
+        for article_block in articles_blocks:
+            try:
+                not_full_url = article_block.find_parent()['href']
+                not_full_urls.append(not_full_url)
+            except TypeError:
+                continue
+
+        for url in not_full_urls:
+            if len(self.urls) < self.total_max_articles and url:
+                full_url = HTTP_PATTERN + url
+                self.urls.append(full_url)
+
+    def find_articles(self):
+        """
+        Finds articles
+        """
+        for seed_url in self.seed_urls:
+            sleep(random.randint(1, 7))
+            response = requests.get(url=seed_url)
+            response.encoding = 'utf-8'
+            if not response.ok:
+                continue
+            soup = BeautifulSoup(response.text, 'lxml')
+            self._extract_url(soup)
+
+    def get_search_urls(self):
+        """
+        Returns seed_urls param
+        """
+        pass
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
+
+    def _fill_article_with_meta_information(self, article_bs):
+        try:
+            self.article.author = article_bs.find('div', class_='article-authors__info').text.strip()
+        except AttributeError:
+            self.article.author = 'NOT FOUND'
+        if self.article.author == 'NOT FOUND':
+            try:
+                self.article.author = article_bs.find('div', class_='article-authors__author').text.strip()
+            except AttributeError:
+                self.article.author = 'NOT FOUND'
+
+        self.article.topics = article_bs.find('span', class_='tags').text.strip().split(' /  ')
+
+        raw_date = article_bs.find('time').text

```

Comment:
can you propose a more simplified solution here? the idea is to avoid so many conditions